src/github/dto/update.practices.dto.ts

```diff
@@ -0,0 +1,11 @@
+import { IsNotEmpty } from 'class-validator';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UpdatePracticesDto {
+  @ApiProperty({
+    description: 'Practices can be strings or a single paragraph.',
+    example: ['practice1', 'practice2'],
+  })
+  @IsNotEmpty()
+  practices: string[];
```

Comment:
PullProofAI

The `@IsNotEmpty()` decorator should be used with `{ each: true }` to validate each element in the array individually.
```diff
-   @IsNotEmpty()
+   @IsArray()
+   @IsNotEmpty({ each: true })
```


<!-- This is an auto-generated comment by PullProofAI -->